src/content/blog/fastify.mdx

```diff
@@ -0,0 +1,179 @@
+---
+title: "Revamping our API pt. 1: Moving from express to Fastify"
+description: How switching to Fastify let us embrace runtime and compile-time types
+pubDate: June 28, 2024
+author: Tom MacWright
+---
+import NpmChart from './fastify/NpmChart.astro';
+
+The JavaScript ecosystem has changed a lot in the last 15 years.
+We've gone from CommonJS to ESM, from browserify to webpack to vite,
+from mocha to vitest, from the request module to using web-standard
+fetch everywhere. The constant reinvention can
+be exciting or annoying from one day to the next.
+
+### The success of Express
+
+In the midst of all this churn, [express][express] has persevered.
+Express is a web framework for Node.js, introduced
+by [TJ Holowaychuk][tj] in 2009. Like a lot
+of early JavaScript modules and many of TJ's projects, it took inspiration
+from the land of Ruby - in particular, [Sinatra][sinatra], which
+was established in 2007.
+
+```js title="express-example.js"
+// This is a tiny example of using
+// express. Notice how there's almost no boilerplate
+// involved in creating a tiny web server.
+const app = require('express')();
+app.get('/', (req, res) => {
+  res.send('Hello World!');
+});
+app.listen(3000);
+```
+
+Express is a beautifully light abstraction: a
+'hello world' example is barely over 10 lines. Compared to the boilerplate
+required to do the same in Ruby on Rails or a Java server at the time, it
+was a breath of fresh air.
+
+<NpmChart />
```

Comment:
Interesting! Why this vs an image?

Reply:
Primarily because I want to to look good on both mobile and desktop, and this does that nicely instead of making text too tiny. And besides that, will probably have charts in future blog posts and wanted to get a handle on what the patterns were.